src/main/java/io/jenkins/pluginhealth/scoring/probes/JenkinsfileProbe.java

```diff
@@ -0,0 +1,45 @@
+package io.jenkins.pluginhealth.scoring.probes;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.stream.Stream;
+
+import io.jenkins.pluginhealth.scoring.model.Plugin;
+import io.jenkins.pluginhealth.scoring.model.ProbeResult;
+
+import org.springframework.core.annotation.Order;
+import org.springframework.stereotype.Component;
+
+@Component
+@Order(value = JenkinsfileProbe.ORDER)
+public class JenkinsfileProbe extends Probe {
+    public static final int ORDER = LastCommitDateProbe.ORDER + 100;
+
+    @Override
+    protected ProbeResult doApply(Plugin plugin, ProbeContext context) {
+        final Path repository = context.getScmRepository();
+        try (Stream<Path> paths = Files
+            .find(repository, 1, (file, basicFileAttributes) -> Files.isReadable(file) && "Jenkinsfile".equals(file.getFileName().toString()))
```

Comment:
Nit: I would search for `Jenkinsfile*`, there are a few exceptions.

Reply:
I know that there are google plugins (ex: https://github.com/jenkinsci/google-kubernetes-engine-plugin) which have a `Jenkinsfile.google` used for their infra. If that is the only `Jenkinsfile` present on the repository, do we still want to consider those plugins as ok?

Reply:
Do you have a lot of example? Because when I run this code, I found that 714 plugins don't have `Jenkinsfile` and when I change it to find files starting with `Jenkinsfile` in their name, I found that 712 plugins don't have that. So, no big difference really..

Reply:
As we have some `Jenkinsfile_k8s` and other exotic filenames in the jenkins-infra org, hence my suggestion, but it seems we are the exception.
We can resolve this conversation, ignore my suggestion please.